src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -442,31 +454,79 @@ export class InstructorCoursesPageComponent implements OnInit {
   /**
    * Copies a feedback session.
    */
-  private copyFeedbackSession(fromFeedbackSession: FeedbackSession, newCourseId: string, oldCourseId: string):
-      Observable<FeedbackSession> {
-    return this.feedbackSessionsService
-      .createFeedbackSession(newCourseId, this.toFbSessionCreationReqWithName(fromFeedbackSession, oldCourseId));
+  private copyFeedbackSession(fromFeedbackSession: FeedbackSession, newCourseId: string,
+                              newTimeZone: string, oldCourseId: string): Observable<FeedbackSession> {
+    return this.feedbackSessionsService.createFeedbackSession(newCourseId,
+        this.toFbSessionCreationReqWithName(fromFeedbackSession, newTimeZone, oldCourseId));
   }
 
   /**
    * Creates a FeedbackSessionCreateRequest with the provided name.
    */
-  private toFbSessionCreationReqWithName(fromFeedbackSession: FeedbackSession, oldCourseId: string):
-      FeedbackSessionCreateRequest {
+  private toFbSessionCreationReqWithName(fromFeedbackSession: FeedbackSession, newTimeZone: string,
+                                         oldCourseId: string): FeedbackSessionCreateRequest {
+    // Local constants
+    const threeHoursBeforeNow = moment().tz(newTimeZone).subtract(3, 'hour').valueOf();
+    const twoDaysFromNowRoundedUp = moment().tz(newTimeZone).add(2, 'days').startOf('hour')
+        .valueOf();
+    const sevenDaysFromNowRoundedUp = moment().tz(newTimeZone).add(7, 'days').startOf('hour')
+        .valueOf();
+    const ninetyDaysFromNow = moment().tz(newTimeZone).add(90, 'day')
+        .valueOf();
+    const oneHundredAndEightyDaysFromNow = moment().tz(newTimeZone).add(180, 'day')
+        .valueOf();
+
+    // Preprocess timestamps to adhere to feedback session timestamps constraints
+    let isModified: boolean = false;
+    let copiedSubmissionStartTimestamp = fromFeedbackSession.submissionStartTimestamp;
+    if (copiedSubmissionStartTimestamp < threeHoursBeforeNow || copiedSubmissionStartTimestamp > ninetyDaysFromNow) {
+      copiedSubmissionStartTimestamp = twoDaysFromNowRoundedUp;
```

Comment:
If the original timestamp is too far, the adjustment should be to the furthest possible that we permit, i.e. (now + 90 days). Adjusting it to (now + 2 days) is a very big surprise.